tests/unit/plugins/modules/test_rds_instance_info.py

```diff
@@ -0,0 +1,121 @@
+# (c) 2022 Red Hat Inc.
+#
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from ansible_collections.amazon.aws.plugins.modules import rds_instance_info
+from unittest.mock import MagicMock, Mock, patch, ANY
+import ansible.module_utils.basic
+import botocore.exceptions
+import pytest
+
+
+mod_name = "ansible_collections.amazon.aws.plugins.modules.rds_instance_info"
+
+
+def a_boto_exception():
+    return botocore.exceptions.UnknownServiceError(
+        service_name="Whoops", known_service_names="Oula"
+    )
+
+
+@patch(mod_name + "._describe_db_instances")
+@patch(mod_name + ".get_instance_tags")
+def test_instance_info_one_instance(m_get_instance_tags, m_describe_db_instances):
+    conn = MagicMock()
+    instance_name = "my-instance"
+    m_get_instance_tags.return_value = []
+    m_describe_db_instances.return_value = [
+        {
+            "DBInstanceIdentifier": instance_name,
+            "DBInstanceArn": "arn:aws:rds:us-east-2:123456789012:og:" + instance_name,
+        }
+    ]
+    rds_instance_info.instance_info(conn, instance_name, filters={})
+
+    m_describe_db_instances.assert_called_with(conn, DBInstanceIdentifier=instance_name)
+    m_get_instance_tags.assert_called_with(
+        conn, arn="arn:aws:rds:us-east-2:123456789012:og:" + instance_name
+    )
+
+
+@patch(mod_name + "._describe_db_instances")
+@patch(mod_name + ".get_instance_tags")
+def test_instance_info_all_instances(m_get_instance_tags, m_describe_db_instances):
+    conn = MagicMock()
+    m_get_instance_tags.return_value = []
+    m_describe_db_instances.return_value = [
+        {
+            "DBInstanceIdentifier": "first-instance",
+            "DBInstanceArn": "arn:aws:rds:us-east-2:123456789012:og:first-instance",
+        },
+        {
+            "DBInstanceIdentifier": "second-instance",
+            "DBInstanceArn": "arn:aws:rds:us-east-2:123456789012:og:second-instance",
+        },
+    ]
+    rds_instance_info.instance_info(conn, instance_name=None, filters={"engine": "postgres"})
+
+    m_describe_db_instances.assert_called_with(conn, Filters=[{'Name': 'engine', 'Values': ['postgres']}])
+    assert m_get_instance_tags.call_count == 2
+    m_get_instance_tags.assert_called_with(
+        conn, arn="arn:aws:rds:us-east-2:123456789012:og:second-instance"
+    )
```

Comment:
```suggestion
    m_get_instance_tags.assert_called_with(
        conn, arn="arn:aws:rds:us-east-2:123456789012:og:first-instance"
    )
    m_get_instance_tags.assert_called_with(
        conn, arn="arn:aws:rds:us-east-2:123456789012:og:second-instance"
    )
```

Reply:
I asume that if we had 2 calls and the second call is the one we expect, then the first call was also ok.

Reply:
Addressed by https://github.com/ansible-collections/amazon.aws/pull/1132/commits/e9a5551c8975db1b288c177de1f4bb3d20b3f36f